Makefile

```diff
@@ -284,7 +284,7 @@ ifdef LUA
     # Windows expects to have lua unpacked at a specific location
     LDFLAGS += -llua
   else
-    LUA_CANDIDATES = lua5.2 lua-5.2 lua5.1 lua-5.1 lua
+    LUA_CANDIDATES = $(basename $(LUA_BINARY)) lua5.2 lua-5.2 lua5.1 lua-5.1 lua
```

Comment:
It is a real problem since it links against random lua libraries which has nothing to do with LUA_BINARY you set.